### PR TITLE
Reduce3 position change

### PR DESCRIPTION
**Pull request by agibsonccc**
Allows for x,y or y,x auto broadcasting on vector/matrix use cases.

### CODE REVIEW

File: tests_cpu/CMakeLists.txt [hunk 1]
```diff
@@ -2,5 +2,5 @@ cmake_minimum_required(VERSION 3.9)
 project(tests_cpu)
 add_subdirectory(lib/googletest-release-1.8.0)
 set(gtest_SOURCE_DIR lib/googletest-release-1.8.0)
-#add_subdirectory(libnd4j_tests)
-add_subdirectory(layers_tests)
\ No newline at end of file
+add_subdirectory(libnd4j_tests)
+#add_subdirectory(layers_tests)
```
Comment (raver119) [inline tests_cpu/CMakeLists.txt]: this change can't be merged to master. tests should be just copied to one folder etc.

File: tests_cpu/layers_tests/CMakeLists.txt [hunk 2]
```diff
@@ -80,7 +80,7 @@ add_library(nd4jcpu STATIC ../../blas/cpu/NativeOps.cpp ../../blas/cpu/GraphExec
         ../../include/cnpy/cnpy.cpp  ../../include/nd4jmemset.h ../../include/nd4jmalloc.h
         ../../blas/Environment.cpp ../../blas/Environment.h
         ${MEMORY_SOURCES} ${GRAPH_SOURCES} ${CUSTOMOPS_SOURCES} ${INDEXING_SOURCES} ${HELPERS_SOURCES} ${CUSTOMOPS_HELPERS_SOURCES}
-        ${OPS_SOURCES} ${LOOPS_SOURCES} ${ARRAY_SOURCES} ${TYPES_SOURCES})
+        ${OPS_SOURCES} ${LOOPS_SOURCES} ${ARRAY_SOURCES} ${TYPES_SOURCES} ../libnd4j_tests/Reduce3Tests.cpp)
```
Comment (raver119) [inline tests_cpu/layers_tests/CMakeLists.txt]: Same as above. Once tests are merged, this thing should be just put into `runtests` section

File: tests_cpu/CMakeLists.txt (hunk 1 above)
Comment (agibsonccc) [reply tests_cpu/CMakeLists.txt]: Just throwing this out there, if  we're intending on this being universal maybe change the names around? Beyond that I agree though.

File: tests_cpu/layers_tests/CMakeLists.txt (hunk 2 above)
Comment (agibsonccc) [reply tests_cpu/layers_tests/CMakeLists.txt]: So "as is" I found a  lot of things don't work out of the box for running tests. Instead, we should get rid of the cause of me needing to do this in the first place.

Comment (raver119) [reply tests_cpu/layers_tests/CMakeLists.txt]: let's just combine these two folders, and both problems will be solved :)

Comment (agibsonccc) [reply tests_cpu/layers_tests/CMakeLists.txt]: K so override it and I'll pull it in to my branch and merge it then update.

Comment (raver119) [reply tests_cpu/layers_tests/CMakeLists.txt]: maybe i'll just update your branch instead?

**Pull request merged**